src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -175,7 +175,7 @@ class WorldConfiguration(@PublishedApi internal val world: World) {
             systemCfg?.invoke(it)
             // assign world systems afterward to resize the systems array only once to the correct size
             // instead of resizing every time a system gets added to the configuration
-            world.systems = it.systems.toTypedArray()
+            world.systems = it.systems
```

Comment:
Since we don't use an array anymore we can directly add the system to `world.systems`. I think a separate world configuration system list is no longer necessary. Also, the comment above is no longer valid and can be removed.

With that change you should be able to avoid `_systems` and `systems` in the world and also replace var with val.